### PR TITLE
Split CMakeLists files and remove unused deps and old ADM files

### PR DESCRIPTION
**Pull request by BrianSipos**
This is not a change in source, just in build config and removing unused content.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ cmake_minimum_required(VERSION 3.10)
 option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
 option(BUILD_DOCS "Enable autodoc building" OFF)
 option(BUILD_AGENT "Build the Agent library and executable" ON)
-option(BUILD_MANAGER "Build the Manager library and executable" ON)
+option(BUILD_MANAGER "Build the Manager library and executable" OFF) # FIXME default to ON
```
Comment (justinethier) [inline CMakeLists.txt:23]: **Question**: Do we want to create an issue to track this or is a FIXME good enough?

File: deps.sh [hunk 2]
```diff
@@ -30,7 +30,7 @@ BUILDDIR=${BUILDDIR:-${SELFDIR}/deps/build}
 echo "Building in ${BUILDDIR}"
 echo "Installing to ${DESTDIR}"
 
-if [ ! -e ${DESTDIR}/usr/include/ion.h ]
+if [ false && ! -e ${DESTDIR}/usr/include/ion.h ]
```
Comment (justinethier) [inline deps.sh:33]: **Question**: Is this intentional and do we want a comment explaining why?

File: CMakeLists.txt (hunk 1 above)
Comment (BrianSipos) [reply CMakeLists.txt:23]: There will be several tickets after the apl-fy24 merge, updating the manager will be one.

File: deps.sh (hunk 2 above)
Comment (BrianSipos) [reply deps.sh:33]: For the moment this project no longer has a build dependence on ION and it's a slow CI step, but there should be a ticket to re-integrate with ION BP transport after apl-fy24 merge.

File: CMakeLists.txt (hunk 1 above)
Comment (BrianSipos) [reply CMakeLists.txt:23]: This is captured in #68

**Pull request merged**